building faileds on shogun repositions.

**Describe the bug**
A clear and concise description of what the bug is.
simple install shogun

c:\vcpkg>vcpkg install shogun:x64-windows-static
Computing installation plan...
The following packages will be built and installed:
    shogun[core]:x64-windows-static
Detecting compiler hash for triplet x64-windows-static...
Starting package 1/1: shogun:x64-windows-static
Building package shogun[core]:x64-windows-static...
Could not locate cached archive: C:\Users\Administrator\AppData\Local\vcpkg/arch
ives\bb\bba3c5a4f94c4ec3944d5f5ecd8846096657135a.zip
-- Using cached C:/vcpkg/downloads/shogun-toolbox-shogun-ab274e7ab6bf24dd598c1da
f1e626cb686d6e1cc.tar.gz
-- Cleaning sources at C:/vcpkg/buildtrees/shogun/src/b686d6e1cc-464b41c80d.clea
n. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/shogun-toolbox-shogun-ab274e7ab6bf24dd59
8c1daf1e626cb686d6e1cc.tar.gz
-- Applying patch cmake.patch
-- Applying patch cmake-config.in.patch
-- Applying patch fix-dirent.patch
-- Using source at C:/vcpkg/buildtrees/shogun/src/b686d6e1cc-464b41c80d.clean
-- Configuring x64-windows-static
-- Building x64-windows-static-dbg
-- Restarting Build without parallelism because memory exceeded
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:141 (message):
    Command failed: C:/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-w
in32-x86/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory: C:/vcpkg/buildtrees/shogun/x64-windows-static-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\shogun\install-x64-windows-static-dbg-out.log
      C:\vcpkg\buildtrees\shogun\install-x64-windows-static-dbg-out-1.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/shogun/portfile.cmake:47 (vcpkg_install_cmake)
  scripts/ports.cmake:79 (include)


Error: Building package shogun:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: shogun:x64-windows-static
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.


**Environment**
- OS: win7 
- Compiler: revision

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install shogun:x64-windows-static
2. See error
Repro code when

**Expected behavior**
A clear and concise description of what you expected to happen.

**Failure logs** 
-(please attached failure logs)

            _Rep=__int64
        ]
C:\vcpkg\installed\x64-windows-static\include\eigen3\Eigen\src/Core/MatrixBase.h(276): fatal error C1060: compiler is out of heap space
[260/659] C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -IC:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src -Isrc -IC:\vcpkg\installed\x64-windows-static\include -IC:\vcpkg\installed\x64-windows-static\include\eigen3 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1    /std:c++14 /bigobj /showIncludes /Fosrc\shogun\CMakeFiles\libshogun.dir\mathematics\linalg\backend\eigen\BasicOps.cpp.obj /Fdsrc\shogun\CMakeFiles\libshogun.dir\ /FS -c C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\mathematics\linalg\backend\eigen\BasicOps.cpp
FAILED: src/shogun/CMakeFiles/libshogun.dir/mathematics/linalg/backend/eigen/BasicOps.cpp.obj 
C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -IC:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src -Isrc -IC:\vcpkg\installed\x64-windows-static\include -IC:\vcpkg\installed\x64-windows-static\include\eigen3 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1    /std:c++14 /bigobj /showIncludes /Fosrc\shogun\CMakeFiles\libshogun.dir\mathematics\linalg\backend\eigen\BasicOps.cpp.obj /Fdsrc\shogun\CMakeFiles\libshogun.dir\ /FS -c C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\mathematics\linalg\backend\eigen\BasicOps.cpp
C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun/lib/parameter_observers/ObservedValue.h(190): warning C4244: 'return': conversion from '_Rep' to 'double', possible loss of data
        with
        [
            _Rep=__int64
        ]
C:\vcpkg\installed\x64-windows-static\include\eigen3\Eigen\src/Core/VectorwiseOp.h(255): fatal error C1060: compiler is out of heap space
[261/659] C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -IC:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src -Isrc -IC:\vcpkg\installed\x64-windows-static\include -IC:\vcpkg\installed\x64-windows-static\include\eigen3 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1    /std:c++14 /bigobj /showIncludes /Fosrc\shogun\CMakeFiles\libshogun.dir\mathematics\linalg\backend\eigen\EigenSolvers.cpp.obj /Fdsrc\shogun\CMakeFiles\libshogun.dir\ /FS -c C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\mathematics\linalg\backend\eigen\EigenSolvers.cpp
C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun/lib/parameter_observers/ObservedValue.h(190): warning C4244: 'return': conversion from '_Rep' to 'double', possible loss of data
        with
        [
            _Rep=__int64
        ]
[262/659] C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -IC:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src -Isrc -IC:\vcpkg\installed\x64-windows-static\include -IC:\vcpkg\installed\x64-windows-static\include\eigen3 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1    /std:c++14 /bigobj /showIncludes /Fosrc\shogun\CMakeFiles\libshogun.dir\mathematics\linalg\backend\eigen\Solvers.cpp.obj /Fdsrc\shogun\CMakeFiles\libshogun.dir\ /FS -c C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\mathematics\linalg\backend\eigen\Solvers.cpp
C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun/lib/parameter_observers/ObservedValue.h(190): warning C4244: 'return': conversion from '_Rep' to 'double', possible loss of data
        with
        [
            _Rep=__int64
        ]
[263/659] C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -IC:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src -Isrc -IC:\vcpkg\installed\x64-windows-static\include -IC:\vcpkg\installed\x64-windows-static\include\eigen3 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1    /std:c++14 /bigobj /showIncludes /Fosrc\shogun\CMakeFiles\libshogun.dir\mathematics\linalg\backend\eigen\Decompositions.cpp.obj /Fdsrc\shogun\CMakeFiles\libshogun.dir\ /FS -c C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\mathematics\linalg\backend\eigen\Decompositions.cpp
C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun/lib/parameter_observers/ObservedValue.h(190): warning C4244: 'return': conversion from '_Rep' to 'double', possible loss of data
        with
        [
            _Rep=__int64
        ]
ninja: build stopped: subcommand failed.


[1/400] cmd.exe /C "cd /D C:\vcpkg\buildtrees\shogun\x64-windows-static-dbg\src\shogun && C:\vcpkg\downloads\tools\cmake-3.17.2-windows\cmake-3.17.2-win32-x86\bin\cmake.exe -D SRC=C:/vcpkg/buildtrees/shogun/src/b686d6e1cc-464b41c80d.clean/src/shogun/lib/versionstring.h.in -D DST=C:/vcpkg/buildtrees/shogun/x64-windows-static-dbg/src/shogun/lib/versionstring.h -D ROOT_DIR=C:/vcpkg/buildtrees/shogun/src/b686d6e1cc-464b41c80d.clean -D PYTHON_EXECUTABLE=C:/vcpkg/downloads/tools/python/python-3.8.3-x64/python.exe -P C:/vcpkg/buildtrees/shogun/src/b686d6e1cc-464b41c80d.clean/cmake/version.cmake"
[2/400] C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -IC:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src -Isrc -IC:\vcpkg\installed\x64-windows-static\include -IC:\vcpkg\installed\x64-windows-static\include\eigen3 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1    /std:c++14 /bigobj /showIncludes /Fosrc\shogun\CMakeFiles\libshogun.dir\base\Parallel.cpp.obj /Fdsrc\shogun\CMakeFiles\libshogun.dir\ /FS -c C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\base\Parallel.cpp
FAILED: src/shogun/CMakeFiles/libshogun.dir/base/Parallel.cpp.obj 
C:\PROGRA~2\MIB055~1\2019\COMMUN~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -IC:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src -Isrc -IC:\vcpkg\installed\x64-windows-static\include -IC:\vcpkg\installed\x64-windows-static\include\eigen3 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1    /std:c++14 /bigobj /showIncludes /Fosrc\shogun\CMakeFiles\libshogun.dir\base\Parallel.cpp.obj /Fdsrc\shogun\CMakeFiles\libshogun.dir\ /FS -c C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\base\Parallel.cpp
C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\base\Parallel.cpp(89): error C3861: 'ASSERT': identifier not found
C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\base\Parallel.cpp(91): error C2146: syntax error: missing ';' before identifier 'num_threads'
ninja: build stopped: subcommand failed.


**Additional context**
Add any other context about the problem here.

Probably due to insufficient memory. The first line of log "fatal error C1060: compiler is out of heap space".
As mentioned in #9588, the libshogun.lib is amazingly huge (dbg: 2.05 GB, rel: 1.80 GB).
I haven't used Shogun for a while, and I don't know if this information is still valid. This information may be useful to you.

> 
> 
> Probably due to insufficient memory. The first line of log "fatal error C1060: compiler is out of heap space".
> As mentioned in #9588, the libshogun.lib is amazingly huge (dbg: 2.05 GB, rel: 1.80 GB).
> I haven't used Shogun for a while, and I don't know if this information is still valid. This information may be useful to you.

but there is an error notice here:

C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\base\Parallel.cpp(89): error C3861: 'ASSERT': identifier not found
C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\base\Parallel.cpp(91): error C2146: syntax error: missing ';' before identifier 'num_threads'

seems my memory is enoght....

Hi @whidbey 

`C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\base\Parallel.cpp(89): error C3861: 'ASSERT': identifier not found`

For this error, you can add `#include <shogun/io/SGIO.h>` in `C:\vcpkg\buildtrees\shogun\src\b686d6e1cc-464b41c80d.clean\src\shogun\base\Parallel.cpp`, then try to build again.

--- a/ports/shogun/CONTROL
+++ b/ports/shogun/CONTROL
@@ -1,6 +1,6 @@
 Source: shogun
 Version: 6.1.4
-Port-Version: 2
+Port-Version: 3
 Build-Depends: bzip2, eigen3, liblzma, libxml2, openblas, nlopt, rxcpp, snappy, zlib, protobuf, curl, lzo, dirent
 Homepage: https://github.com/shogun-toolbox/shogun
 Description: Unified and efficient Machine Learning

--- a/ports/shogun/fix-ASSERT-not-found.patch
+++ b/ports/shogun/fix-ASSERT-not-found.patch
@@ -1,0 +1,12 @@
+diff --git a/src/shogun/base/Parallel.cpp b/src/shogun/base/Parallel.cpp
+index 78ba319..53ac6c0 100644
+--- a/src/shogun/base/Parallel.cpp
++++ b/src/shogun/base/Parallel.cpp
+@@ -12,6 +12,7 @@
+ #include <shogun/lib/RefCount.h>
+ #include <shogun/lib/config.h>
+ #include <shogun/lib/memory.h>
++#include <shogun/io/SGIO.h>
+ 
+ #ifdef HAVE_CXX11
+ #include <thread>

--- a/ports/shogun/portfile.cmake
+++ b/ports/shogun/portfile.cmake
@@ -10,6 +10,7 @@ vcpkg_from_github(
         cmake.patch
         cmake-config.in.patch
         fix-dirent.patch
+        fix-ASSERT-not-found.patch
 )
 
 vcpkg_find_acquire_program(PYTHON3)

